### PR TITLE
use prefetched user list in select user component

### PR DESCRIPTION
**Pull request by petrjasek**
there was an issue when fetching desk members
for a desk with many users, it would fail on
too big request when fetching 50+ users by id.

SDCP-849

### CODE REVIEW

**Review by thecalcc — APPROVED** [review]

File: scripts/api/index.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import {time} from './time';
 import {user} from './user';
 import {vocabularies} from './vocabularies';
 import {contentProfiles} from './content-profiles';
+import * as utils from './utils';
```
Comment (tomaskikutis) [inline scripts/api/index.ts]: why not import from lodash directly?

File: scripts/core/ui/components/SelectUser.tsx [hunk 2]
```diff
@@ -118,47 +117,23 @@ export class SelectUser extends SuperdeskReactComponent<IPropsSelectUser, IState
...
-
-                    if (this.props.deskId != null && this.props.deskId != '') {
-                        query.$and.push({_id: {$in: deskMemberIds}});
-                    }
+                    const deskMembers = sdApi.desks.getDeskMembers(this.props.deskId);
 
                     if (searchString != null && searchString.length > 0) {
-                        query.$and.push(getUserSearchMongoQuery(searchString));
-                    }
+                        const query = getUserSearchMongoQuery(searchString);
```
Comment (tomaskikutis) [inline scripts/core/ui/components/SelectUser.tsx]: Your filtering here only supports a subset of mongo query, ($or operator). How about next to `getUserSearchMongoQuery` you put `matchUserByMongoQuery(user: IUser): boolean` so if someone changes the query they would also see client side matching code and would update it too.

**Comment by tomaskikutis** [comment]
> too big request when fetching 50+ users by id

You mean GET request URL would get too long?

**Comment by tomaskikutis** [comment]
By the way, that playwright test is unstable. You can either try restarting e2e or disable that test. (the problem is actually the feature that is unstable, not the test)

**Comment by petrjasek** [comment]
> > too big request when fetching 50+ users by id
> 
> You mean GET request URL would get too long?

yes exactly

File: scripts/api/index.ts (hunk 1 above)
Comment (petrjasek) [reply scripts/api/index.ts]: will drop it, seems like we have another helper for it

File: scripts/core/ui/components/SelectUser.tsx (hunk 2 above)
Comment (petrjasek) [reply scripts/core/ui/components/SelectUser.tsx]: done

**Review by tomaskikutis — APPROVED** [review]

**Comment by tomaskikutis** [comment]
I was looking around the code and realized this change is breaking `SelectUser` interface and probably usages. According to the interface `deskId` is optional and according to your code it looks like it wouldn't work without it?

**Comment by petrjasek** [comment]
yep you're right, will fix it

**Pull request merged**